SCClassLibrary/Common/GUI/PlusGUI/Control/server-scope.sc

```diff
@@ -1,16 +1,35 @@
 + Server {
-	scope { arg numChannels, index = 0, bufsize = 4096, zoom = (1), rate = \audio;
+	scope { arg numChannels, index = 0, bufsize = 4096, zoom = (1), rate = \audio, bounds = nil;
 		numChannels = numChannels ?? { if (index == 0) { options.numOutputBusChannels } { 2 } };
 
-		if(scopeWindow.isNil) {
-			scopeWindow = Stethoscope(this, numChannels, index, bufsize, zoom, rate, nil,
-				this.options.numBuffers);
+		if(bounds.isNil) {
+			if(scopeWindow.isNil) {
+				scopeWindow = Stethoscope(this, numChannels, index, bufsize, zoom, rate, nil,
+					this.options.numBuffers);
 				// prevent buffer conflicts by using reserved bufnum
-			scopeWindow.window.onClose = scopeWindow.window.onClose.addFunc({ scopeWindow = nil });
+				scopeWindow.window.onClose = scopeWindow.window.onClose.addFunc({ scopeWindow = nil });
+			} {
+				scopeWindow.setProperties(numChannels, index, bufsize, zoom, rate);
+				scopeWindow.run;
+				scopeWindow.window.front;
+			}
```

Comment:
Looks like you could avoid the duplicated code and extra logic caused by `if(bounds.isNil)` by letting `scopeWindow` be created as before, then if the `bounds` arg is provided, access the window enclosing `scopeWindow` with `scopeWindow.window.bounds_(bounds)`. (untested)

That way responsibility for creating the window stays in one place.

Reply:
Thanks, done!
https://github.com/supercollider/supercollider/pull/6230#issuecomment-1987051825